### PR TITLE
462 sigel process new

### LINKED ISSUE #462: Update Sigil process
We are using an old dump from 2013. This makes the transformation pretty pretty slow, since it harvests all updates since then via OAI-PMH.
We have a new one bit it is binary pica+. We need to remodel it.

- [x] Change the process. #472
- [ ] ~Add cron job that looks for new updated batch file. That downloads the file to `app/transformation/input`, renames the file to `sigel.dat` and cjanges the date in `transformation.updates.start="2023-06-01"` to the creation date of the sourcefile in `conf/application.conf`~

### PR DESCRIPTION
**Pull request by TobiasNx**
Resolves #462

I tried to setup to change to code so that the pica binary file is read instead of the sigil.xml

This PR does three things:
- Uses a Pica-Dump instead of the old XML-Dump.
- Splits the process for the piccadata into pica binary dump transformation and OAI-PMH Pica XML Update transformation
- Updates the tests

~~This process seems to have some error, since I do not know when to close the streams, if at all.~~
Could you have a look and adjust the workflows since I have no Java skills.

I updated the testfile to a pica binary file with 6 entries, using the same ISILs that were used in the old metadata plus more. I updated the transformation-tests ~but the other play test still need some fixing~. Also the play tests are updated. (Edit: 10.08.23)

______________________

Old workflow (xml dump and lots of oai updates) needs 3 1/2 min to transform and index:
```
2023-08-08 15:43:05 +0200 [INFO] from application in
				main - Starting transformation, will write to '/tmp/lobid-organisations/enriched.prod.out.json' 
2023-08-08 15:46:39 +0200 [INFO] from play in
				main - Application started (Prod)
2023-08-08 15:46:39 +0200 [INFO] from play in
				main - Listening for HTTP on /0:0:0:0:0:0:0:0:9000
```

New process (pica binary dump + small number of oai updates) takes 1 min to transform and index:
```
2023-08-10 13:22:18 +0200 [INFO] from application in
				main - Starting transformation, will write to '/tmp/lobid-organisations/enriched.prod.out.json' 
2023-08-10 13:23:33 +0200 [INFO] from play in
				main - Application started (Prod)
2023-08-10 13:23:34 +0200 [INFO] from play in
				main - Listening for HTTP on /0:0:0:0:0:0:0:0:9000
```

### CODE REVIEW

File: test/transformation/TestTransformAll.java [hunk 1]
```diff
@@ -87,16 +86,16 @@ public void preferSigelData() throws IOException {
...
+	//@Test
+	//public void sigelSplitting() {
+	//	final FileOpener sourceFileOpener = new FileOpener();
+	//	final XmlElementSplitter xmlSplitter = new XmlElementSplitter(
+	//			TransformSigel.DUMP_TOP_LEVEL_TAG, TransformSigel.DUMP_ENTITY);
+	//	TransformSigel.setupSigelSplitting(sourceFileOpener, xmlSplitter,
+	//			DUMP_XPATH, TransformAll.DATA_OUTPUT_DIR);
+	//	sourceFileOpener.process(SIGEL_DUMP_LOCATION);
+	//	sourceFileOpener.closeStream();
+	//}
```
Comment (TobiasNx) [inline test/transformation/TestTransformAll.java:98]: We could test instead one single day via oai pmh?

File: app/transformation/TransformSigel.java [hunk 2]
```diff
@@ -71,19 +93,21 @@ static void process(String startOfUpdates, int intervalSize,
...
+	// 	final FileOpener dumpFileOpener = new FileOpener();
+	// 	dumpFileOpener//
+	// 			.setReceiver(new XmlDecoder())//
+	// 			.setReceiver(new XmlElementSplitter(DUMP_TOP_LEVEL_TAG, DUMP_ENTITY))//
+	// 			.setReceiver(
+	// 					xmlFilenameWriter(TransformAll.DATA_OUTPUT_DIR, DUMP_XPATH));
+	// 	dumpFileOpener.process(TransformAll.DATA_INPUT_DIR + "sigel.xml");
+	// 	dumpFileOpener.closeStream();
+	// }
+
```
Comment (dr0i) [inline app/transformation/TransformSigel.java:110]: if this is not necessary we should remove the comments

Comment (TobiasNx) [reply app/transformation/TransformSigel.java:110]: +1

File: test/transformation/TestTransformAll.java (hunk 1 above)
Comment (TobiasNx) [reply test/transformation/TestTransformAll.java:98]: Delete comment.

**Review by dr0i — APPROVED** [review]
Improved a lot of minor things.
You can merge.

**Pull request merged**